### PR TITLE
Cvsl 1386 add endpoint security test to probation teams

### PR DESCRIPTION
**Pull request by nishanthmahasamudram**

### CODE REVIEW

File: src/main/java/uk/gov/justice/hmpps/probationteams/controllers/LocalDeliveryUnitController.kt [hunk 1]
```diff
@@ -19,6 +20,7 @@ import uk.gov.justice.hmpps.probationteams.services.LocalDeliveryUnitService
   value = ["local-delivery-units"],
   produces = [APPLICATION_JSON_VALUE],
 )
+@PreAuthorize("hasAnyRole('MAINTAIN_REF_DATA', 'SYSTEM_USER', 'ROLE_SYSTEM_USER')")
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/probationteams/controllers/LocalDeliveryUnitController.kt:23]: hasAnyRole implicitly adds the `ROLE_` prefix so `SYSTEM_USER` and `ROLE_SYSTEM_USER` are the same

But lets just remove both `SYSTEM_USER` and `ROLE_SYSTEM_USER` from each endpoint - other teams are trying to deprecate it.

File: src/main/java/uk/gov/justice/hmpps/probationteams/controllers/ProbationAreaController.kt [hunk 2]
```diff
@@ -28,6 +29,7 @@ import uk.gov.justice.hmpps.probationteams.services.SetOutcome
   value = ["probation-areas"],
   produces = [APPLICATION_JSON_VALUE],
 )
+@PreAuthorize("hasAnyRole('MAINTAIN_REF_DATA', 'SYSTEM_USER', 'ROLE_SYSTEM_USER')")
```
Comment (andrewrlee) [inline src/main/java/uk/gov/justice/hmpps/probationteams/controllers/ProbationAreaController.kt:32]: Can you also remove all the `PreAuthorize` annotations from the service layer?  So we just have it at the controller level?

File: src/main/java/uk/gov/justice/hmpps/probationteams/controllers/LocalDeliveryUnitController.kt (hunk 1 above)
Comment (nishanthmahasamudram) [reply src/main/java/uk/gov/justice/hmpps/probationteams/controllers/LocalDeliveryUnitController.kt:23]: ok I will just leave MAINTAIN_REF_DATA role.

**Review by andrewrlee — APPROVED** [review]
LGTM

**Pull request merged**